### DISCUSSION TITLE
Enable airflow local and remote logging simultanoeusly.

### DISCUSSION THREAD

**Discussion by alfad-pai** [apache/airflow#29724] [Q&A]
Hi.
I have a specific use case where I am pushing airflow logs to AWS S3.However, now I have to push logs to Loki which means I have to write the logs locally as Airflow doesn't write logs in Loki's format.Based on what I understood from the documentation, Airflow either supports remote **or** local logging **but not both** . Is there a way to write logs both remotely and locally?

**Comment by potiuk** [comment]
Generally airflow logs are being written locally as well when you do remote logging (especially with S3 where you have to store logs locally before you push them). But parsing local logs to send them to loki is rather bad idea. Instead you are completely free to modify airflow logging https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html#advanced-configuration -> airflow uses standard Python logging framework to log, so if you develop a Loki Python logging handler, you should be able to add it as a handler to log your messages there.
